### PR TITLE
Add 1.82 release notes

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
cc @rust-lang/release 

This is using the new release notes infrastructure. My expectation is that most and/or all edits should be made separately from this PR by updating various relnote tracking issues, and then we'll re-run the relnotes tool. That probably comes with an exception for the stabilized APIs for std, which will need to be manually added at the end.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,100 @@
...
+
+
+<a id="1.82.0-Libraries"></a>
+
+Libraries
+---------
+- [Implement DoubleEnded and ExactSize for Take<Repeat> and Take<RepeatWith>](https://github.com/rust-lang/rust/pull/106943/)
+- [Stabilize Wasm relaxed SIMD](https://github.com/rust-lang/rust/pull/117468/)
+- [Stabilize const `{integer}::from_str_radix` i.e. `const_int_from_str`](https://github.com/rust-lang/rust/pull/124941/)
+- [CommandExt::before_exec: deprecate safety in edition 2024](https://github.com/rust-lang/rust/pull/125970/)
```
Comment (Noratrieb) [inline RELEASES.md:34]:
```suggestion
- [`CommandExt::before_exec`: deprecate safety in edition 2024](https://github.com/rust-lang/rust/pull/125970/)
```

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,100 @@
...
+
+Cargo
+-----
+- [feat: Add `info` cargo subcommand](https://github.com/rust-lang/cargo/pull/14141/)
+
+<a id="1.82.0-Compatibility-Notes"></a>
+
+Compatibility Notes
+-------------------
+- [`addr_of(_mut)!` macros and the newly stabilized `&raw (const|mut)` are now safe to use with all static items](https://github.com/rust-lang/rust/pull/125834)
```
Comment (Noratrieb) [inline RELEASES.md:54]: this is language

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,100 @@
...
+ - We now [disallow setting some built-in cfgs via the command-line](https://github.com/rust-lang/rust/pull/126158) with the newly added [`explicit_builtin_cfgs_in_flags`](https://doc.rust-lang.org/rustc/lints/listing/deny-by-default.html#explicit-builtin-cfgs-in-flags) lint in order to prevent incoherent state, eg. `windows` cfg active but target is Linux based. The appropriate [`rustc` flag](https://doc.rust-lang.org/rustc/command-line-arguments.html) should be used instead.
+- [Stabilize `&raw const` and `&raw mut` operators (RFC 2582)](https://github.com/rust-lang/rust/pull/127679)
```
Comment (Noratrieb) [inline RELEASES.md:56]: language, not compat

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,100 @@
...
+- [Stabilize unsafe extern blocks (RFC 3484)](https://github.com/rust-lang/rust/pull/127921)
```
Comment (Noratrieb) [inline RELEASES.md:57]: language, not compat

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,100 @@
...
+- The standard library has a new implementation of `binary_search` which is significantly improves performance (#128254). However when a sorted slice has multiple values which compare equal, the new implementation may select a different value among the equal ones than the old implementation.
+- [illumos/Solaris now sets `MSG_NOSIGNAL` when writing to sockets](https://github.com/rust-lang/rust/pull/128259). This avoids killing the process with SIGPIPE when writing to a closed socket, which matches the existing behavior on other UNIX targets.
+- [Stabilize offset_of_nested](https://github.com/rust-lang/rust/pull/128284)
```
Comment (Noratrieb) [inline RELEASES.md:60]: library (?), not compat

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,100 @@
...
+  commonly used instruction by the WebAssembly backend, has changed. Validators
+  and parsers which don't understand the `reference-types` proposal will no
+  longer accept modules produced by LLVM due to this change in encoding of
+  immediates. Additionally these features being enabled are encoded in the
+  `target_features` custom section and may affect downstream tooling such as
+  `wasm-opt` consuming the module. Generating a WebAssembly module that disables
+  default features requires `-Zbuild-std` support from Cargo and more information
+  can be found at
+  [rust-lang/rust#128511](https://github.com/rust-lang/rust/pull/128511).
+- [Stabilize `const` arguments to inline asm](https://github.com/rust-lang/rust/pull/128570)
```
Comment (Noratrieb) [inline RELEASES.md:80]: language (?), not compat

File: RELEASES.md [hunk 7]
```diff
@@ -1,3 +1,100 @@
...
+- [Promote aarch64-apple-darwin to Tier 1](https://github.com/rust-lang/rust/pull/128592)
```
Comment (Noratrieb) [inline RELEASES.md:81]: compiler, not compat

File: RELEASES.md [hunk 8]
```diff
@@ -1,3 +1,100 @@
...
+- [stabilize floating-point arithmetic in `const fn`](https://github.com/rust-lang/rust/pull/128596)
```
Comment (Noratrieb) [inline RELEASES.md:82]: language, not compat

File: RELEASES.md [hunk 9]
```diff
@@ -1,3 +1,100 @@
...
+- [float types: document NaN bit pattern guarantees](https://github.com/rust-lang/rust/pull/129559)
```
Comment (Noratrieb) [inline RELEASES.md:83]: language, not compat

File: RELEASES.md [hunk 10]
```diff
@@ -1,3 +1,100 @@
+Version 1.82.0 (2024-10-17)
+==========================
+
+<a id="1.82.0-Language"></a>
+
+Language
+--------
+- [Don't make statement nonterminals match pattern nonterminals](https://github.com/rust-lang/rust/pull/120221/)
+- [Implement lint against ambiguous negative literals](https://github.com/rust-lang/rust/pull/121364/)
```
Comment (Urgau) [inline RELEASES.md:9]: While this lint was implemented, it was revert to allow-by-default over concern about usability.

@Mark-Simulacrum What should be done about thing that are no longer release notes worthy? Should we close the release note tracking issue?

File: RELEASES.md [hunk 11]
```diff
@@ -1,3 +1,100 @@
...
+
+<a id="1.82.0-Libraries"></a>
+
+Libraries
+---------
+- [Implement DoubleEnded and ExactSize for Take<Repeat> and Take<RepeatWith>](https://github.com/rust-lang/rust/pull/106943/)
+- [Stabilize Wasm relaxed SIMD](https://github.com/rust-lang/rust/pull/117468/)
+- [Stabilize const `{integer}::from_str_radix` i.e. `const_int_from_str`](https://github.com/rust-lang/rust/pull/124941/)
+- [CommandExt::before_exec: deprecate safety in edition 2024](https://github.com/rust-lang/rust/pull/125970/)
+- [CloneToUninit impls](https://github.com/rust-lang/rust/pull/126877/)
```
Comment (jieyouxu) [inline RELEASES.md:35]:
```suggestion
- [`CloneToUninit` impls](https://github.com/rust-lang/rust/pull/126877/)
```

File: RELEASES.md (hunk 10 above)
Comment (Mark-Simulacrum) [reply RELEASES.md:9]: Yes, that should be sufficient to remove the item from this listing (once I rerun the tool). Essentially all the edits proposed here will need to be transferred elsewhere and then ideally are self-resolved by the update.

File: RELEASES.md (hunk 11 above)
Comment (jieyouxu) [reply RELEASES.md:35]: This one is also kinda terse, but I don't understand the context enough to better reflect what changed here. Seems internal though, so I suppose that's fine.

File: RELEASES.md (hunk 10 above)
Comment (Urgau) [reply RELEASES.md:9]: Great, to hear.

Btw, I actually can't find the relnote tracking issue concerning this item #121364, was it pull from one or is there still a fallback from the `finished-final-comment-period` labels.

File: RELEASES.md [hunk 12]
```diff
@@ -1,3 +1,100 @@
...
+- [Improved diagnostics to explain why a pattern is unreachable](https://github.com/rust-lang/rust/pull/128034)
+- [The compiler now triggers the unreachable code warning properly for async functions that don't return/are `-> !`](https://github.com/rust-lang/rust/pull/128443)
+- [Promote `wasm32-wasip2` to Tier 2.](https://github.com/rust-lang/rust/pull/126967/)
+
+
+<a id="1.82.0-Libraries"></a>
+
+Libraries
+---------
+- [Implement DoubleEnded and ExactSize for Take<Repeat> and Take<RepeatWith>](https://github.com/rust-lang/rust/pull/106943/)
```
Comment (slanterns) [inline RELEASES.md:31]:
```suggestion
- [Implement DoubleEnded and ExactSize for `Take<Repeat>` and `Take<RepeatWith>`](https://github.com/rust-lang/rust/pull/106943/)
```

File: RELEASES.md (hunk 11 above)
Comment (lukas-code) [reply RELEASES.md:44]: This allows calling `Rc<T>/Arc<T>::make_mut` for `T` = `Path`, `OsStr`, and `CStr` on stable.

In 1.81 we made a similar change for just `[T]` and `str` and the relnotes list that as

> Generalize `{Rc,Arc}::make_mut()` to unsized types.

so perhaps this one could be

```suggestion
- [Generalize `{Rc,Arc}::make_mut()` to `Path`, `OsStr`, and `CStr`.](https://github.com/rust-lang/rust/pull/126877/)
```

File: RELEASES.md [hunk 13]
```diff
@@ -6,21 +6,31 @@ Version 1.82.0 (2024-10-17)
 Language
 --------
 - [Don't make statement nonterminals match pattern nonterminals](https://github.com/rust-lang/rust/pull/120221/)
-- [Implement lint against ambiguous negative literals](https://github.com/rust-lang/rust/pull/121364/)
 - [Patterns matching empty types can now be omitted in common cases](https://github.com/rust-lang/rust/pull/122792)
+- [`addr_of(_mut)!` macros and the newly stabilized `&raw (const|mut)` are now safe to use with all static items](https://github.com/rust-lang/rust/pull/125834)
 - [size_of_val_raw: for length 0 this is safe to call](https://github.com/rust-lang/rust/pull/126152/)
 - [Reorder trait bound modifiers *after* `for<...>` binder in trait bounds](https://github.com/rust-lang/rust/pull/127054/)
```
Comment (lukas-code) [inline RELEASES.md:12]: We still need to update the reference grammar here: https://doc.rust-lang.org/nightly/reference/trait-bounds.html#trait-and-lifetime-bounds

Comment (Mark-Simulacrum) [reply RELEASES.md:12]: Can you file a separate issue or otherwise track that? This doesn't seem like something relevant to the release notes?

Comment (lukas-code) [reply RELEASES.md:13]: The reference is updated now: https://github.com/rust-lang/reference/pull/1650.

File: RELEASES.md [hunk 14]
```diff
@@ -1,3 +1,104 @@
...
+Cargo
+-----
+- [feat: Add `info` cargo subcommand](https://github.com/rust-lang/cargo/pull/14141/)
+
+<a id="1.82.0-Compatibility-Notes"></a>
+
+Compatibility Notes
+-------------------
+ - We now [disallow setting some built-in cfgs via the command-line](https://github.com/rust-lang/rust/pull/126158) with the newly added [`explicit_builtin_cfgs_in_flags`](https://doc.rust-lang.org/rustc/lints/listing/deny-by-default.html#explicit-builtin-cfgs-in-flags) lint in order to prevent incoherent state, eg. `windows` cfg active but target is Linux based. The appropriate [`rustc` flag](https://doc.rust-lang.org/rustc/command-line-arguments.html) should be used instead.
+- The standard library has a new implementation of `binary_search` which is significantly improves performance (#128254). However when a sorted slice has multiple values which compare equal, the new implementation may select a different value among the equal ones than the old implementation.
```
Comment (theemathas) [inline RELEASES.md:67]: The 128254 should be a proper link

**Pull request merged**